.github/workflows/mac_python_matrix_test.yml

```diff
@@ -0,0 +1,49 @@
+name: MacOS Pyomo Tests Workflow (Slim, Matrixed, Nightly)
+
+on:
+  pull_request:
+    branches:
+      - master
+      # Can add additional branches if desired
+
+jobs:
+  pyomo-mac-tests:
+    name: Pyomo Test (${{ matrix.python-version }}, MacOS)
```

Comment:
Can we just make this `py${{ matrix.python-version }}`?  Combined with the above, that should give us entries in the checks that read `continuous-integration/github/pr/osx / py3.8`

Reply:
Has been added for the Linux and Mac files.